Add external webhook handling function

Rdilshan opened the pull request.

Rdilshan committed:
Add external webhook handling function

Rdilshan committed:
Add webhook handling functions for retrieving webhooks and their secrets

Rdilshan committed:
update to support multiple webhook requests with error handling.

SajeeTechi commented on supabase/functions/_db/webhooks.ts:
@Rdilshan add the return type of the db service functions

SajeeTechi commented on supabase/functions/external-webhooks/index.ts:
```diff
@@ -0,0 +1,91 @@
+  }
+
+  //for get the alias
+  const webhook = data[0];
```
multiple communities (i.e aliases) can subscribe to the same webhook.
you cannot pick the alias of the first item in the array

SajeeTechi commented on supabase/functions/external-webhooks/index.ts:
```diff
@@ -0,0 +1,91 @@
+
+  // Fetch the webhook secret from the database
+  const { data: webhookSecretData, error: webhookSecretError } =
+    await getWebhookSecret(supabaseClient, alias);
```
for each unique alias in webhook array return its webhook secret

SajeeTechi commented on supabase/functions/external-webhooks/index.ts:
```diff
@@ -0,0 +1,91 @@
+  const webhookSecretKey = webhookSecret.secret;
+
+  // Send all webhooks requests
+  for (const webhook of data) {
```
consider above comments before editing the iterator

SajeeTechi commented on supabase/functions/external-webhooks/index.ts:
@Rdilshan comments added per line

Rdilshan committed:
Enhance webhook functions with TypeScript interfaces and return types

SajeeTechi committed:
change to appropriate error message

SajeeTechi reviewed (changes requested):
@Rdilshan Pending PR changes

Rdilshan committed:
Refactor webhook handling

SajeeTechi committed:
refactor(webhooks): improve error handling and logging in external webhooks

- Replace error responses with continue statements to process remaining webhooks
- Add URL information to error and success log messages
- Improve null checking for webhook secret data
- Enhance logging clarity with more detailed webhook status messages

SajeeTechi committed:
packages
